proof/refine/ARM/Finalise_R.thy

```diff
@@ -4016,6 +4013,81 @@ lemma unbind_maybe_notification_corres:
   apply (clarsimp simp: valid_obj'_def valid_ntfn'_def split: option.splits)
   done
 
+lemma update_sched_context_corres:
```

Comment:
This is very similar to `update_sc_no_reply_stack_update_corres`. Did you want to move this lemma to where that one is, and maybe refactor things so that the existing lemma uses this one, if possible?

Reply:
That's where I must have got this lemma! I forgot how this lemma came to be and don't remember writing it at all. I'll move it to `update_sc_no_reply_stack_update_corres` then and refactor.

Reply:
I've moved it there, but couldn't seem to refactor (just tried `corres_symb_exec_r` followed by `update_sched_context_corres` but this put the proof in a weird state for the first assumption). Might try again with someone's help.